src/ZenithAngleCalc.jl

```diff
@@ -1,322 +1,116 @@
-"""
-function GMST(date::DateTime, timezone::FT) where {FT <: Real}
-    jc = julian_century(date)
-    UTC_hours = Dates.hour(date) + Dates.minute(date)/60.0 + Dates.second(date)/3600.0 - timezone
-    UTC_deg = UTC_hours * 15.0
-
-    # Greenwich mean sidereal time, radians
-    GMSTdeg = mod(100.460618375 + 36000.7700536083*jc + 0.00038793*jc^2 - 2.58333e-8*jc^3 + UTC_deg, 360.0)
-    GMSTrad = deg2rad(GMSTdeg)
-
-    return GMSTrad
-end
-
-"""
-    earth_sun_dist(ecc::FT, TA::FT) where {FT <: Real}
-
-returns the Earth-sun distance in meters
-given the eccentricity and true anomaly
-
-formula from "Astronomical Algorithms" by Jean Meeus
-chapter 25, equation 25.5
-"""
-function earth_sun_dist(ecc::FT, TA::FT) where {FT <: Real}
-    d_au = (1.000001018 * (1.0 - ecc^2)) / (1.0 + ecc*cos(TA))
-    d = d_au * astro_unit()
-    return d
+function mean_anomaly_vernal_equinox(ϖ::FT, e::FT) where {FT <: Real}
+    # mean anomaly at vernal equinox, radians (3.11)
+    β = (1-e^2)^0.5
+    M_v = -ϖ + (e+0.25*e^3)*(1+β)*sin(ϖ) - 0.5*e^2*(0.5+β)*sin(2ϖ) + 0.25*e^3*(1/3+β)*sin(3ϖ)
+    return M_v
 end
 
-"""
-    instantaneous_zenith_angle(date::DateTime,
-                               timezone::FT,
-                               longitude::FT,
-                               latitude::FT) where {FT <: Real}
-
-returns the zenith angle and earth-sun distance
-at a particular longitude and latitude on the given date
-
-equations from "Astronomical Algorithms" by Jean Meeus
-see documentation for details
-"""
-function instantaneous_zenith_angle(date::DateTime,
-                                    timezone::FT,
-                                    longitude::FT,
-                                    latitude::FT) where {FT <: Real}
-    λ = deg2rad(longitude)
-    ϕ = deg2rad(latitude)
+function distance_declination(date::DateTime, param_set::APS, γ::FT, ϖ::FT, e::FT) where {FT <: Real}
+    Ya::FT = year_anom(param_set)
+    day_length::FT = Planet.day(param_set)
+    AU::FT = astro_unit()
     
-    TL = true_longitude(date)
-    TA = true_anomaly(date)
-    ecc = eccentricity(date)
-    γ = obliquity(date)
-    GMSTrad = GMST(date, timezone)
-    d = earth_sun_dist(ecc, TA)
+    # time of vernal equinox in the epoch (rearrangement of 3.6 and 3.10)
+    M_v0 = mean_anomaly_vernal_equinox(ϖ_epoch(), e)
+    time_v = Ya * (M_v0 - M_epoch()) / 2π + epoch()
 
-    # declination, radians
-    δ = mod(asin(sin(γ) * sin(TL)), 2*π)
-    # right acension, radians
-    Y = cos(γ) * sin(TL)
-    X = cos(TL)
-    RA = mod(atan(Y,X), 2*π)
+    # mean anomaly given mean anomaly at vernal equinox (3.10)
+    time = datetime2julian(date)*day_length
+    M_v = mean_anomaly_vernal_equinox(ϖ, e)
+    MA = mod(2π * (time - time_v) / Ya + M_v, 2π)
 
-    # hour angle, radians
-    LMST = mod(GMSTrad + λ, 2*π)
-    η = mod(LMST - RA, 2*π)
+    # true anomaly, radians (3.8)
+    TA = mod(MA + (2*e - e^FT(3/4))*sin(MA) + FT(5/4)*e^2*sin(2*MA) + FT(13/12)*e^3*sin(3*MA), 2π)
```

Comment:
```suggestion
    TA = mod(MA + (2*e - e^FT(3)/4)*sin(MA) + FT(5/4)*e^2*sin(2*MA) + FT(13/12)*e^3*sin(3*MA), 2π)
```
(That's may be why you got the error for the rotation of perihelion.) You should also encapsulate constants here and in other formulas in `FT(...)`, and use things such as `0.25 *...` and `.../4` consistently.

Reply:
Yes! Problem solved. Thank you for spotting that typo.

Reply:
Looks good :) 
![insol_example2b](https://user-images.githubusercontent.com/6875824/109763667-a43ebb80-7baf-11eb-8c2b-8d343fb2945c.png)